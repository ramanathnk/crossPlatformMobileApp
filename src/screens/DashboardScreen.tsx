import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  SafeAreaView,
  ScrollView,
} from 'react-native';
import { StatusBar } from 'expo-status-bar';
import SnaptrackerLogo from '../icons/SnapTrackerLogoNew';

import { useNavigation } from '@react-navigation/native';
import type { StackNavigationProp } from '@react-navigation/stack';
import type { RootStackParamList } from '../../App';
import MaterialIconComponent from '../icons/MaterialIconComponent';

import SettingsIcon from '../../assets/material-icons/settings.svg';
import TableEditIcon from '../../assets/material-icons/table_edit.svg';
import AddBoxIcon from '../../assets/material-icons/add_box.svg';
import ConciergeIcon from '../../assets/material-icons/concierge.svg';
import MobileCheckIcon from '../../assets/material-icons/mobile_check.svg';
import PendingActionsIcon from '../../assets/material-icons/pending_actions.svg';
import HomeIcon from '../../assets/material-icons/home.svg';
import GroupIcon from '../../assets/material-icons/group.svg';



const DashboardScreen: React.FC = () => {
  const navigation = useNavigation<StackNavigationProp<RootStackParamList, 'Dashboard'>>();
  return (
    <SafeAreaView style={styles.container}>
      <StatusBar style="light" />
      <ScrollView
        contentContainerStyle={styles.scrollContainer}
        showsVerticalScrollIndicator={false}
      >
        {/* Logo and App Name */}
        <View style={styles.headerContainer}>
          <View style={styles.logoContainer}>
            <SnaptrackerLogo width={120} height={50} />
            <Text style={styles.appName}>SnapTracker</Text>
          </View>
        </View>

        {/* Header */}
        <View style={styles.dashboardHeader}>
          <View style={styles.dashboardTitleContainer}>
            <Text style={styles.dashboardTitle}>Dashboard</Text>
            <Text style={styles.dashboardSubtitle}>Device Registration Portal</Text>
          </View>
          <TouchableOpacity
            style={styles.logoutButton}
            activeOpacity={1}
          >
            <Text style={styles.logoutText}>AT</Text>
          </TouchableOpacity>
        </View>

        {/* Statistics Cards */}
        <View style={styles.statsContainer}>
          <View style={styles.statCard}>
            <View style={styles.statIconContainer}>
              <MaterialIconComponent Icon={PendingActionsIcon} size={24} color="#fff" />
            </View>
            <Text style={styles.statNumber}>12</Text>
            <Text style={styles.statLabel}>Pending Requests</Text>
          </View>

          <View style={styles.statCard}>
            <View style={styles.statIconContainer}>
              <MaterialIconComponent Icon={MobileCheckIcon} size={24} color="#fff" />
            </View>
            <Text style={styles.statNumber}>12</Text>
            <Text style={styles.statLabel}>Registered</Text>
          </View>
        </View>

        {/* Core Modules */}
        <View style={styles.sectionContainer}>
          <Text style={styles.sectionTitle}>CORE MODULES</Text>
          <View style={styles.moduleGrid}>
            <TouchableOpacity style={styles.moduleCard}>
              <View style={styles.moduleIconContainer}>
                <MaterialIconComponent Icon={MobileCheckIcon} size={24} color="#fff" />
              </View>
              <Text style={styles.moduleTitle}>Device Requests</Text>
              <Text style={styles.moduleSubtitle}>Review & approve</Text>
            </TouchableOpacity>

            <TouchableOpacity style={styles.moduleCard} onPress={() => navigation.navigate('RegisterDevice')}>
              <View style={styles.moduleIconContainer}>
                <MaterialIconComponent Icon={AddBoxIcon} size={24} color="#fff" />
              </View>
              <Text style={styles.moduleTitle}>Register Device</Text>
              <Text style={styles.moduleSubtitle}>Add new device</Text>
            </TouchableOpacity>

            <TouchableOpacity style={styles.moduleCard}>
              <View style={styles.moduleIconContainer}>
                <MaterialIconComponent Icon={TableEditIcon} size={24} color="#fff" />
              </View>
              <Text style={styles.moduleTitle}>Manage Records</Text>
              <Text style={styles.moduleSubtitle}>Edit & update</Text>
            </TouchableOpacity>

            <TouchableOpacity style={styles.moduleCard}>
              <View style={styles.moduleIconContainer}>
                <MaterialIconComponent Icon={TableEditIcon} size={24} color="#fff" />
              </View>
              <Text style={styles.moduleTitle}>View Devices</Text>
              <Text style={styles.moduleSubtitle}>All Registrations</Text>
            </TouchableOpacity>
          </View>
        </View>

        {/* Recent Activity */}
        <View style={styles.sectionContainer}>
          <View style={styles.sectionHeader}>
            <Text style={styles.sectionTitle}>RECENT ACTIVITY</Text>
            <TouchableOpacity>
              <Text style={styles.viewAllText}>View all</Text>
            </TouchableOpacity>
          </View>
          <View style={styles.activityList}>
            <Text style={styles.noActivityText}>No recent activity</Text>
          </View>
        </View>

        {/* Bottom Navigation */}
        <View style={styles.bottomNav}>
          <TouchableOpacity style={styles.navItem}>
            <MaterialIconComponent Icon={HomeIcon} size={24} color="#9CA3AF" />
            <Text style={styles.navLabel}>Dashboard</Text>
          </TouchableOpacity>

          <TouchableOpacity style={styles.navItem}>
            <MaterialIconComponent Icon={ConciergeIcon} size={24} color="#9CA3AF" />
            <Text style={styles.navLabel}>Requests</Text>
          </TouchableOpacity>

          <TouchableOpacity style={styles.navItem}>
            <MaterialIconComponent Icon={GroupIcon} size={24} color="#9CA3AF" />
            <Text style={styles.navLabel}>Dealers</Text>
          </TouchableOpacity>

          <TouchableOpacity style={styles.navItem}>
            <MaterialIconComponent Icon={SettingsIcon} size={24} color="#9CA3AF" />
            <Text style={styles.navLabel}>Settings</Text>
          </TouchableOpacity>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#1F2937',
  },
  scrollContainer: {
    flexGrow: 1,
    paddingHorizontal: 24,
    paddingTop: 60,
    paddingBottom: 40,
  },
  headerContainer: {
    alignItems: 'center',
    marginBottom: 40,
  },
  logoContainer: {
    alignItems: 'center',
  },
  logo: {
    width: 60,
    height: 60,
    backgroundColor: '#3B82F6',
    borderRadius: 12,
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 12,
  },
  logoText: {
    fontSize: 28,
    color: '#FFFFFF',
  },
  appName: {
    fontSize: 24,
    fontWeight: '600',
    color: '#FFFFFF',
    marginTop: 8,
  },
  dashboardHeader: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    marginBottom: 24,
    paddingHorizontal: 4,
  },
  dashboardTitleContainer: {
    flex: 1,
  },
  dashboardTitle: {
    fontSize: 24,
    fontWeight: '700',
    color: '#FFFFFF',
    marginBottom: 4,
  },
  dashboardSubtitle: {
    fontSize: 14,
    color: '#9CA3AF',
  },
  logoutButton: {
    width: 40,
    height: 40,
    backgroundColor: '#374151',
    borderRadius: 8,
    justifyContent: 'center',
    alignItems: 'center',
  },
  logoutText: {
    fontSize: 16,
    fontWeight: '600',
    color: '#FFFFFF',
  },
  statsContainer: {
    flexDirection: 'row',
    gap: 16,
    marginBottom: 32,
  },
  statCard: {
    flex: 1,
    backgroundColor: '#374151',
    borderRadius: 12,
    padding: 16,
    alignItems: 'center',
  },
  statIconContainer: {
    width: 40,
    height: 40,
    backgroundColor: '#4B5563',
    borderRadius: 8,
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 12,
  },
  statIconGreen: {
    backgroundColor: '#059669',
  },
  statIcon: {
    fontSize: 20,
    color: '#FFFFFF',
  },
  statNumber: {
    fontSize: 24,
    fontWeight: '700',
    color: '#FFFFFF',
    marginBottom: 4,
  },
  statLabel: {
    fontSize: 12,
    color: '#9CA3AF',
    textAlign: 'center',
  },
  sectionContainer: {
    marginBottom: 32,
  },
  sectionHeader: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 16,
  },
  sectionTitle: {
    fontSize: 12,
    fontWeight: '600',
    color: '#9CA3AF',
    letterSpacing: 1,
  },
  viewAllText: {
    fontSize: 14,
    color: '#3B82F6',
  },
  moduleGrid: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    gap: 16,
  },
  moduleCard: {
    width: '47%',
    backgroundColor: '#374151',
    borderRadius: 12,
    padding: 16,
    alignItems: 'center',
  },
  moduleIconContainer: {
    width: 48,
    height: 48,
    backgroundColor: '#4B5563',
    borderRadius: 12,
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 12,
  },
  moduleIcon: {
    fontSize: 24,
    color: '#FFFFFF',
  },
  moduleTitle: {
    fontSize: 14,
    fontWeight: '600',
    color: '#FFFFFF',
    marginBottom: 4,
    textAlign: 'center',
  },
  moduleSubtitle: {
    fontSize: 12,
    color: '#9CA3AF',
    textAlign: 'center',
  },
  activityList: {
    backgroundColor: '#374151',
    borderRadius: 12,
    padding: 20,
    alignItems: 'center',
  },
  noActivityText: {
    fontSize: 14,
    color: '#9CA3AF',
  },
  bottomNav: {
    flexDirection: 'row',
    backgroundColor: '#374151',
    borderRadius: 16,
    paddingVertical: 12,
    paddingHorizontal: 8,
    marginTop: 'auto',
  },
  navItem: {
    flex: 1,
    alignItems: 'center',
    paddingVertical: 8,
  },
  navIcon: {
    fontSize: 20,
    marginBottom: 4,
  },
  navLabel: {
    fontSize: 10,
    color: '#9CA3AF',
  },
  navLabelActive: {
    color: '#3B82F6',
  },
});

export default DashboardScreen;
