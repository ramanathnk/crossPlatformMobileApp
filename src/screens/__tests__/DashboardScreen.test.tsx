import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import DashboardScreen from '../DashboardScreen';

describe('DashboardScreen', () => {
  it('renders the dashboard title', () => {
    const { getByText } = render(
      <NavigationContainer>
        <DashboardScreen />
      </NavigationContainer>
    );
    expect(getByText('Dashboard')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const { toJSON } = render(
      <NavigationContainer>
        <DashboardScreen />
      </NavigationContainer>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});