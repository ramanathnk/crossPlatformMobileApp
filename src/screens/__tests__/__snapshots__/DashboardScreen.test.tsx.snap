// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardScreen matches snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#1F2937",
      "flex": 1,
      "paddingBottom": 8,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "flexGrow": 1,
        "paddingBottom": 40,
        "paddingHorizontal": 24,
        "paddingTop": 40,
      }
    }
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 8,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <Image
              resizeMode="contain"
              source="FileMock"
              style={
                {
                  "height": 32,
                  "width": 32,
                }
              }
            />
          </View>
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontSize": 20,
                "fontWeight": "600",
                "marginLeft": 10,
              }
            }
          >
            SnapTracker
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#374151",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
              "opacity": 1,
              "width": 40,
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            AT
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "marginBottom": 24,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 24,
              "fontWeight": "700",
              "marginBottom": 4,
            }
          }
        >
          Dashboard
        </Text>
        <Text
          style={
            {
              "color": "#9CA3AF",
              "fontSize": 14,
            }
          }
        >
          Device Registration Portal
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "gap": 16,
            "justifyContent": "space-between",
            "marginBottom": 32,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#2D3748",
              "borderRadius": 16,
              "elevation": 2,
              "flex": 1,
              "marginHorizontal": 4,
              "padding": 20,
              "shadowColor": "#000",
              "shadowOpacity": 0.08,
              "shadowRadius": 8,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 12,
                  "height": 48,
                  "justifyContent": "center",
                  "marginBottom": 12,
                  "width": 48,
                },
                {
                  "backgroundColor": "#F59E42",
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  {
                    "height": 28,
                    "width": 28,
                  },
                ]
              }
            >
              <svg
                fill="#fff"
                height={28}
                width={28}
              />
            </View>
          </View>
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontSize": 24,
                "fontWeight": "700",
                "marginBottom": 4,
              }
            }
          >
            12
          </Text>
          <Text
            style={
              {
                "color": "#9CA3AF",
                "fontSize": 12,
                "textAlign": "center",
              }
            }
          >
            Pending Requests
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#2D3748",
              "borderRadius": 16,
              "elevation": 2,
              "flex": 1,
              "marginHorizontal": 4,
              "padding": 20,
              "shadowColor": "#000",
              "shadowOpacity": 0.08,
              "shadowRadius": 8,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 12,
                  "height": 48,
                  "justifyContent": "center",
                  "marginBottom": 12,
                  "width": 48,
                },
                {
                  "backgroundColor": "#10B981",
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  {
                    "height": 28,
                    "width": 28,
                  },
                ]
              }
            >
              <svg
                fill="#fff"
                height={28}
                width={28}
              />
            </View>
          </View>
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontSize": 24,
                "fontWeight": "700",
                "marginBottom": 4,
              }
            }
          >
            7
          </Text>
          <Text
            style={
              {
                "color": "#9CA3AF",
                "fontSize": 12,
                "textAlign": "center",
              }
            }
          >
            Registered Today
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "marginBottom": 32,
          }
        }
      >
        <Text
          style={
            {
              "color": "#9CA3AF",
              "fontSize": 12,
              "fontWeight": "600",
              "letterSpacing": 1,
            }
          }
        >
          CORE MODULES
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "gap": 16,
              "justifyContent": "space-between",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#374151",
                "borderRadius": 12,
                "marginBottom": 16,
                "opacity": 1,
                "padding": 16,
                "width": "47%",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#4B5563",
                  "borderRadius": 12,
                  "height": 48,
                  "justifyContent": "center",
                  "marginBottom": 12,
                  "width": 48,
                }
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    {
                      "height": 28,
                      "width": 28,
                    },
                  ]
                }
              >
                <svg
                  fill="#3B82F6"
                  height={28}
                  width={28}
                />
              </View>
            </View>
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "marginBottom": 4,
                  "textAlign": "center",
                }
              }
            >
              Device Requests
            </Text>
            <Text
              style={
                {
                  "color": "#9CA3AF",
                  "fontSize": 12,
                  "textAlign": "center",
                }
              }
            >
              Review & approve
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#374151",
                "borderRadius": 12,
                "marginBottom": 16,
                "opacity": 1,
                "padding": 16,
                "width": "47%",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#4B5563",
                  "borderRadius": 12,
                  "height": 48,
                  "justifyContent": "center",
                  "marginBottom": 12,
                  "width": 48,
                }
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    {
                      "height": 28,
                      "width": 28,
                    },
                  ]
                }
              >
                <svg
                  fill="#3B82F6"
                  height={28}
                  width={28}
                />
              </View>
            </View>
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "marginBottom": 4,
                  "textAlign": "center",
                }
              }
            >
              Register Device
            </Text>
            <Text
              style={
                {
                  "color": "#9CA3AF",
                  "fontSize": 12,
                  "textAlign": "center",
                }
              }
            >
              Add new device
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#374151",
                "borderRadius": 12,
                "marginBottom": 16,
                "opacity": 1,
                "padding": 16,
                "width": "47%",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#4B5563",
                  "borderRadius": 12,
                  "height": 48,
                  "justifyContent": "center",
                  "marginBottom": 12,
                  "width": 48,
                }
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    {
                      "height": 28,
                      "width": 28,
                    },
                  ]
                }
              >
                <svg
                  fill="#3B82F6"
                  height={28}
                  width={28}
                />
              </View>
            </View>
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "marginBottom": 4,
                  "textAlign": "center",
                }
              }
            >
              Manage Records
            </Text>
            <Text
              style={
                {
                  "color": "#9CA3AF",
                  "fontSize": 12,
                  "textAlign": "center",
                }
              }
            >
              Edit & update
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#374151",
                "borderRadius": 12,
                "marginBottom": 16,
                "opacity": 1,
                "padding": 16,
                "width": "47%",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#4B5563",
                  "borderRadius": 12,
                  "height": 48,
                  "justifyContent": "center",
                  "marginBottom": 12,
                  "width": 48,
                }
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    {
                      "height": 28,
                      "width": 28,
                    },
                  ]
                }
              >
                <svg
                  fill="#3B82F6"
                  height={28}
                  width={28}
                />
              </View>
            </View>
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "marginBottom": 4,
                  "textAlign": "center",
                }
              }
            >
              View Devices
            </Text>
            <Text
              style={
                {
                  "color": "#9CA3AF",
                  "fontSize": 12,
                  "textAlign": "center",
                }
              }
            >
              All registrations
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "marginBottom": 32,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#9CA3AF",
                "fontSize": 12,
                "fontWeight": "600",
                "letterSpacing": 1,
              }
            }
          >
            RECENT ACTIVITY
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#3B82F6",
                  "fontSize": 14,
                }
              }
            >
              View all
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#374151",
              "borderRadius": 12,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              {
                "color": "#9CA3AF",
                "fontSize": 14,
              }
            }
          >
            No recent activity
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
